core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -168,6 +170,26 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
           driverConf.set(key, value)
         }
       }
+      // Delegation Token Updater is not supported in Hadoop 1, so use reflection.
+      // Can't use Option[ExecutorDelegationTokenUpdater] because it is built only in YARN
+      // profile, so use Option[Any] since even the stop method call will be via reflection.
+      var tokenUpdaterOption: Option[Any] = None
+      var tokenUpdaterClass: Option[Class[_]] = None
```

Comment:
We can keep the reflection out if we separate it out into different methods - that seems like a sensible thing to do, that way all of the runtime-related error-prone code can be gotten rid of. 

I will update this with something that does that.